test/hdt-test.js

```diff
@@ -115,6 +115,50 @@ describe('hdt', function () {
         );
       });
     });
+
+    describe('fetching distinct terms', function () {
+      it('Should have correct results for given object', function () {
+        return document.searchTerms({ position : 'predicate', filter : { value : 'http://example.org/o001', position : 'object' } }).then(terms => {
```

Comment:
Only object is tested. Either we should also test other cases, or remove them from the code (and throw an error).

Reply:
We should test other positions indeed. I'd say, test whether a fixed sub or pred term throw the appropriate error